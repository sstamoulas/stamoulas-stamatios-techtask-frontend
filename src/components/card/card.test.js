import React from 'react'
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
 
import Card from './card.component'
 
const mockStore = configureStore([])
const recipes = [{
  "title": "Ham and Cheese Toastie",
  "ingredients": [
    "Ham",
    "Cheese",
    "Bread",
    "Butter"
  ]
},
{
  "title": "Salad",
  "ingredients": [
    "Lettuce",
    "Tomato",
    "Cucumber",
    "Beetroot",
    "Salad Dressing"
  ]
}]

describe('Card Component', () => {
  let store
  let component
 
  beforeEach(() => {
    store = mockStore({
      recipes: {
        recipes,
      },
    })
 
    component = create(
      <Provider store={store}>
        <Card />
      </Provider>
    )
  })
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
