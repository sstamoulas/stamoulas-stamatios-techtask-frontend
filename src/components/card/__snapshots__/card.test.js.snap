// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render with given state from Redux store 1`] = `
<div
  className="card-container"
>
  <div
    className="flip-card"
  >
    <div
      className="flip-card-inner"
    >
      <div
        className="flip-card-front"
      >
        <div>
          Ham and Cheese Toastie
        </div>
      </div>
      <div
        className="flip-card-back"
      >
        <div>
          Ingredients:
        </div>
        <ul
          className="list-container"
        >
          <li
            className="card-list-item"
          >
            Ham
          </li>
          <li
            className="card-list-item"
          >
            Cheese
          </li>
          <li
            className="card-list-item"
          >
            Bread
          </li>
          <li
            className="card-list-item"
          >
            Butter
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="flip-card"
  >
    <div
      className="flip-card-inner"
    >
      <div
        className="flip-card-front"
      >
        <div>
          Salad
        </div>
      </div>
      <div
        className="flip-card-back"
      >
        <div>
          Ingredients:
        </div>
        <ul
          className="list-container"
        >
          <li
            className="card-list-item"
          >
            Lettuce
          </li>
          <li
            className="card-list-item"
          >
            Tomato
          </li>
          <li
            className="card-list-item"
          >
            Cucumber
          </li>
          <li
            className="card-list-item"
          >
            Beetroot
          </li>
          <li
            className="card-list-item"
          >
            Salad Dressing
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
